src/lib/io/include/sol-iio.h

```diff
@@ -77,6 +77,7 @@ typedef struct sol_iio_config {
     const void *data; /**< User defined data to be sent to sol_iio_reader_cb */
```

Comment:
Well, as you are changing a versioned structure, I'd say that `SOL_IIO_CONFIG_API_VERSION` may need to be changed - even if only adding stuff. @barbieri ?